content/02-understand-prisma/05-under-the-hood.mdx

```diff
@@ -51,3 +51,86 @@ A Prisma engine is the **direct interface to the database**, any higher-level in
 As an example, Prisma Client connects to the [query engine](../reference/tools-and-interfaces/prisma-client/query-engine) in order to read and write data in a database:
 
 ![](https://i.imgur.com/Nwwx5Cv.png)
+
+### Using custom engine binaries
+
+By default, all binaries are automatically downloaded into the `node_modules\@prisma\cli` folder when you install or update `@prisma\cli`. You might want to use a custom binary file if:
+
+* Automated download of binaries is not possible.
+* You have created your own engine binary (for testing, or for an OS that is not officially supported).
+
+Use the following environment variables to specify custom locations for your binaries:
+
+* `PRISMA_MIGRATION_ENGINE_BINARY` (Migration engine)
+* `PRISMA_QUERY_ENGINE_BINARY` (Query engine)
+* `PRISMA_INTROSPECTION_ENGINE_BINARY` (Introspection engine)
+* `PRISMA_FMT_BINARY` (for `npx prisma format`)
+
+
+#### The `prisma\.env` file
+
+To set an environment variable in the `prisma\.env` file:
+
+1. Add the environment variable to the `prisma\.env` file:
+
+   ```
+   PRISMA_QUERY_ENGINE_BINARY=c:\custom\my-query-engine-windows.exe
+   ```
+
+2. Run the following command to output the paths to all binaries:
+
+   ```bash-symbol
+   npx prisma -v
+   ```
+
+   The output shows that the query engine path comes from the `PRISMA_QUERY_ENGINE_BINARY` environment variable:
+
+   ```bash-symbol
+   Current platform     : windows
+   + Query Engine         : query-engine d6ff7119649922b84e413b3b69660e2f49e2ddf3 (at c:\custom\my-query-engine-windows.exe)
```

Comment:
Not sure why there is a + 🤔

Reply:
This gets turned into a diff in the visual output. Not sure if that is the right thing to do here vs. highlighting the line (which I think our docs can also do)

Reply:
I do not know why I put a `+` instead of a `|`. 🙃 #BrainFog